### PR TITLE
Add support for JWT authorization

### PR DESCRIPTION
**Pull request by savonarola**
Now clients may be authorized with respect to a specific claim containing publish/subscribe topic whitelists.

### CODE REVIEW

File: apps/emqx_auth_jwt/src/emqx_auth_jwt_svr.erl [hunk 1]
```diff
@@ -181,6 +165,26 @@ cancel_timer(State = #state{tref = TRef}) ->
...
+                   _ -> keys(static)
+               end,
+        case Jwks of
+            [] -> {error, not_found};
+            _ ->
+                do_verify(JwsCompacted, Jwks)
+        end
+    catch
+        Class : Reason : Stk ->
+            ?LOG(error, "Handle JWK crashed: ~p, ~p, stacktrace: ~p~n",
```
Comment (thalesmg) [inline apps/emqx_auth_jwt/src/emqx_auth_jwt_svr.erl:183]: Nit:

```suggestion
            ?LOG(error, "do_verify JWK crashed: ~p, ~p, stacktrace: ~p~n",
```

File: apps/emqx_auth_jwt/test/emqx_auth_jwt_SUITE.erl [hunk 2]
```diff
@@ -164,3 +168,129 @@ t_check_claims_kid_in_header(_) ->
     Result0 = emqx_access_control:authenticate(Plain#{password => Jwt}),
     ct:pal("Auth result: ~p~n", [Result0]),
     ?assertMatch({ok, #{auth_result := success, jwt_claims := _}}, Result0).
+
+t_check_jwt_acl(_) ->
+    application:set_env(emqx_auth_jwt, verify_claims, [{sub, <<"value">>}]),
+    application:stop(emqx_auth_jwt), application:start(emqx_auth_jwt),
```
Comment (thalesmg) [inline apps/emqx_auth_jwt/test/emqx_auth_jwt_SUITE.erl:174]: Maybe it's safer to do this setup (and later clean-up) in `{init,end}_per_testcase`.

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — COMMENTED** [review]
Just a reminder: might need to update the changelog.

File: apps/emqx_auth_jwt/test/emqx_auth_jwt_SUITE.erl (hunk 2 above)
Comment (savonarola) [reply apps/emqx_auth_jwt/test/emqx_auth_jwt_SUITE.erl]: ✅

File: apps/emqx_auth_jwt/src/emqx_auth_jwt_svr.erl (hunk 1 above)
Comment (savonarola) [reply apps/emqx_auth_jwt/src/emqx_auth_jwt_svr.erl]: ✅

File: apps/emqx_auth_jwt/etc/emqx_auth_jwt.conf [hunk 3]
```diff
@@ -47,3 +47,8 @@ auth.jwt.verify_claims = off
 ## For example, to verify that the username in the JWT payload is the same
 ## as the client (MQTT protocol) username
 #auth.jwt.verify_claims.username = %u
+
+## Name of the clame containg ACL rules
```
[the remark this answers is not in the document]
Comment (savonarola) [reply apps/emqx_auth_jwt/etc/emqx_auth_jwt.conf]: ✅

File: apps/emqx_auth_jwt/README.md [hunk 4]
```diff
@@ -46,6 +46,11 @@ auth.jwt.verify_claims = off
 ##  - %u: username
 ##  - %c: clientid
 # auth.jwt.verify_claims.username = %u
+
+## Name of the clame containg ACL rules
```
[the remark this answers is not in the document]
Comment (savonarola) [reply apps/emqx_auth_jwt/README.md]: ✅

**Review by ieQu1 — APPROVED** [review]

File: CHANGES-4.3.md [hunk 5]
```diff
@@ -32,6 +32,8 @@ File format:
   scenarios, especially when the emqx-bridge-mqtt plugin is configured as shared
   subscription. [#7462]
 * Add some compression functions to rule-engine: gzip, gunzip, zip, unzip, zip_compress, zip_uncompress
+* Add support for JWT authorization [#7596]
```
Comment (zmstone) [inline CHANGES-4.3.md:35]: Should have been a part of 4.3.15

**Pull request merged**